### PR TITLE
Issue/#387 validate coop results before adding to leaderboard

### LINKED ISSUE #387: Coop leaderboards doesn't validate game settings
Report from the forums: https://forums.faforever.com/viewtopic.php?f=3&t=17070

The server should check if the game was played without any sim mods and with defaults settings. Else the replay should not be added into the leaderboards.

### PR DESCRIPTION
**Pull request by Askaholic**
Fixes #387 

* [x] add test

Would be nice if there was a catch all 'Invalid Game Setting' validity state.

Requires:
https://github.com/FAForever/db/pull/187

### CODE REVIEW

File: server/games/game.py [hunk 1]
```diff
@@ -540,39 +542,76 @@ def clear_slot(self, slot_index):
...
+                    return False
+        return True
 
-        elif self.gameOptions["NoRushOption"] != "Off":
-            await self.mark_invalid(ValidityState.NORUSH_ENABLED)
+    async def _validate_coop_game_settings(self):
+        """
+        Checks which only apply to the coop mode
+        """
+        # Note, the UNEVEN_TEAMS_NOT_RANKED state is used as a placeholder for reasons
```
Comment (Rackover) [inline server/games/game.py]: This sounds really dirty to me. I would prefer you to add all of those  conditions in the DB via a migration scripts, and to add them to server code later.

File: server/games/game.py [hunk 2]
```diff
@@ -87,6 +87,10 @@ class ValidityState(IntEnum):
     UNLOCKED_TEAMS = 17
     MULTI_TEAM = 18
     HAS_AI_PLAYERS = 19
+    CIVILIANS_REVEALED = 20
+    TOO_EASY = 21
```
Comment (Rackover) [inline server/games/game.py]: WRONG_DIFFICULTY please 😛

Comment (Askaholic) [reply server/games/game.py]: Ok. Was trying to be consistent, but wrong difficulty is probably better.

**Comment by Rackover** [comment]
Tested, working, merged.

**Pull request merged**